src/wakefs/daemon_client.cpp

```diff
@@ -34,9 +34,16 @@
 #include "json/json5.h"
 #include "fuse.h"
 
+// The user-id and group-id are used so that fuse daemons with different uid:gid pairs
+// running within the same build can co-exist without trying to share. The kernel
+// prevents cross-user sharing when the 'allow_other' fuse mount option is not used.
+const std::string mount_path_id(const std::string& base_dir) {
+    return base_dir + "/.fuse-" + std::to_string(getuid()) + ":" + std::to_string(getgid());
```

Comment:
If we care about Windows compatibility, colons aren't allowed in filenames and we might want to choose a different separator.  Since Wake isn't tested on Windows that may be an unnecessary concern, but I just wanted to mention it anyway.

Reply:
ah ok, can change it

Reply:
Totally unrelated to this review that's an interesting question I've been wondering. If we get wake into a better state it might be useful for a much wider audience but most large projects that might use wake (and then contribute back) require the big three to all work (linux, mac, windows). If supporting mac/windows becomes a real burden then we should make a choice about cutting support.

For now I suspect that just sandboxing less as needed is a viable mechanism to support mac/windows so simple things like this leave open the opportunity for future open source success.

Reply:
I don't think we try to support windows in general, just for the vscode plugin

Reply:
Though I guess if someone were willing to put in the work for it

Reply:
I would propose "/.fuse/{uid}.{gid}" so that all the fuse stuff remains in a directory called '.fuse', because some other parts of the codebase in wake give that directory special treatment.

Reply:
Currently, wake runs natively on OS/X and runs in WSL under windows.

Reply:
yep can try that out instead

Reply:
> Totally unrelated to this review that's an interesting question I've been wondering. If we get wake into a better state it might be useful for a much wider audience but most large projects that might use wake (and then contribute back) require the big three to all work (linux, mac, windows). If supporting mac/windows becomes a real burden then we should make a choice about cutting support.
> 
> For now I suspect that just sandboxing less as needed is a viable mechanism to support mac/windows so simple things like this leave open the opportunity for future open source success.



> 
any way to fix  error that no fuse in wsl  when compiling wake src code？

Reply:
Please to not resurrect old pull requests to ask questions. Use the Github Issues mechanism instead.